### ISSUE TITLE
Scrolling on second monitor in Firefox and iTunes

### ISSUE THREAD

**Issue by ttencate** [ttencate/taekwindow#53]
Converted from [SourceForge issue 2314897](http://sourceforge.net/support/tracker.php?aid=2314897), submitted by nobody

Hi Thomas,

When I select the option 'Scroll the window under the cursor', scrolling won't work in a firefox tab, even when I select the tab. Hope you can fix this.

Samuel

**Comment by ttencate** [comment]
Submitted by nobody

Sorry, I cannot reproduce this behaviour. Scrolling in Firefox works perfectly for me, and for everyone I know that uses Taekwindow. If it were consistently broken, I'm sure I would have had a bug report before.

That being said, there is clearly something wrong if it does not work for you. Since I can't reproduce this isuue, I would need your help to fix it. I could send you a debug version. However, I currently don't have time for any of this, so it will have to wait at least a couple of weeks. If you could send me an e-mail in the meantime, or reply to this bugreport under your SourceForge account if you have one, then I will have a means of contacting you, and I'll get in touch as soon as I have time!

**Comment by ttencate** [comment]
Submitted by nobody

Finally narrowed this issue down! If Firefox or iTunes is on the secondary monitor, and the secondary monitor is to the left of the primary, then this problem occurs consistently.

Probably something to do with negative coordinates. Will look into this!

**Comment by ttencate** [comment]
Submitted by nobody

I do have the same problem. Secondary monitor on the left and no scrolling in firefox. What a pity.

**Comment by ttencate** [comment]
Submitted by nobody

// BEGIN HACK for Firefox.
// Firefox won't receive correctly scrolling messages when on a screen on the left
// of the main screen
bool isFirefoxWindow(HWND window) {
    return windowHasClass(window, L"MozillaWindowClass");
}
// END HACK

bool doMouseWheel(HWND window, POINT mousePos, WPARAM wParam) {
    HWND targetWindow = WindowFromPoint(mousePos);

```
// BEGIN HACK for mIRC
if (isMIRCWindow(targetWindow)) {
    targetWindow = findMIRCTextWindow(targetWindow);
}
// END HACK
// BEGIN HACK for Firefox.
if( mousePos.x < 0 && isFirefoxWindow(targetWindow) )
{
    SetForegroundWindow(targetWindow);
    return false;
}
// END HACK
```

Tadaaa!

**Comment by ttencate** [comment]
Submitted by nobody

Well, that might work, but will activate the Firefox window too... and worse, bring it to the front of the several dozen chat windows that you have hovering over it... which mouse wheel events normally don't do. But apparently you do not mind, so you're happy :)

I might use your Firefox detection code, though. Completely drowning in work right now, will not have time to look into this until next month.

**Comment by ttencate** [comment]
Submitted by nobody

Probable cause is this line in actions.cpp:

LPARAM lParam = ((short)(mousePos.x)) | (((short)(mousePos.y)) << 16);

**Comment by ttencate** [comment]
Submitted by nobody

I am having this problem as well.  Not just with firefox, either...  Version 0.2.4 of taekwindow is extremely flaky about scrolling windows on my left-of-primary second display.  It's not limited strictly to firefox, either...  I've also seen the issue in chrome windows and in miro (xulrunner-based, i think) and perhaps in the windows explorer as well...  

pretty much it happens with ANY WiNDOW [or subwindow] THAT DOES NOT HAVE FOCUS....  the mouse wheel will still scroll a window in the foreground, but taekwindow cannot steer events to scrollable panes in a focused window or scroll visible regions of buried windows.

**Comment by ttencate** [comment]
Submitted by nobody

If a window has focus, then it will scroll....or at least the portion of the window that has focus.  

but if it's buried, or there are several panes in the window, then taekwindow cannot steer the wheel events to the appropriate scrollable area....at least on the left screen...but taekwindow can scroll partially hidden windows and unfocused subwindows on the main screen...

**Comment by ttencate** [comment]
Submitted by nobody

Indeed a cast/bitshift issue. Fixed in 0.3, to be released very soon.
